ros/src/computing/perception/detection/lidar_tracker/packages/lidar_imm_ukf_pda_track/nodes/imm_ukf_pda/imm_ukf_pda.cpp

```diff
@@ -752,6 +767,42 @@ void ImmUkfPda::pubDebugRosMarker(const autoware_msgs::DetectedObjectArray& inpu
   pub_texts_array_.publish(texts_markers);
 }
 
+void ImmUkfPda::dumpResultText(autoware_msgs::DetectedObjectArray& detected_objects)
+{
+  std::ofstream outputfile(result_file_path_, std::ofstream::out | std::ofstream::app);
+  for(size_t i = 0; i < detected_objects.objects.size(); i++)
+  {
+    tf::Quaternion q(detected_objects.objects[i].pose.orientation.x, detected_objects.objects[i].pose.orientation.y,
```

Comment:
You can abbreviate the code by using tf::quaternionMsgToTF in "tf/transform_datatypes.h".
[http://docs.ros.org/melodic/api/tf/html/c++/transform__datatypes_8h.html](url)